code/src/main/java/com/onlinecontacttracing/servlets/authentication/CheckForApiAuthorizationServlet.java

```diff
@@ -0,0 +1,125 @@
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Collections;
+import java.util.List;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.auth.oauth2.AuthorizationRequestUrl;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.FileInputStream;
+import java.io.File;
+import java.security.GeneralSecurityException;
+import java.io.FileNotFoundException;
+import java.util.logging.Logger;
+
+/**
+*  This class directs the creation of user Credentials for accessing APIs.
+*/
+
+@WebServlet("/check-for-api-authorization")
+public abstract class CheckForApiAuthorizationServlet extends HttpServlet {
+
+  abstract void useCredential(Credential credential);
+  // should return "check-for-api-authorization" for now
+  abstract String getServletURIName();
+  //Update the userId with the newly created credential
+  abstract void updateUser(String userId);
+
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final List<String> SCOPES = Collections.singletonList("https://www.googleapis.com/auth/contacts.readonly");
+  private static final String CREDENTIALS_FILE_PATH = "WEB-INF/credentials.json";
+  private static final String url = "https://covid-catchers-fixed-gcp.ue.r.appspot.com";
+  static final Logger log = Logger.getLogger(CheckForApiAuthorizationServlet.class.getName());
+
+  //Creates the user's credential
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      // Get flow for token response
+      GoogleAuthorizationCodeFlow flow = getFlow(response);
+      String code = request.getParameter("code");
+      String idTokenString = request.getParameter("state");
+
+      // Get userId
+      String userId = getUserId(idTokenString, flow, response);
+      TokenResponse tokenResponse = flow.newTokenRequest(code).setRedirectUri(url+getServletURIName()).execute();
+      Credential credential = flow.createAndStoreCredential(tokenResponse, userId);
+      updateUser(userId);
+      useCredential(credential);
+    } catch (Exception e) {
```

Comment:
So, this will swallow all exceptions, but only handle the two listed below.

You probably want to do

try {...}
catch (FileNotFoundException e) { ... }
catch (GeneralSecurityException e) {...}
catch (Exception e) {...}

They are evaluated in order of block here. You want the most general last. If you only want to handle the two exceptions listed here, and want any other to get passed further up, don't add the Exception e block.